### PR TITLE
OAK-10093 : provided support for SSE Customer Keys for AWS

### PR DESCRIPTION
**Pull request by rishabhdaim**

### CODE REVIEW

**Review by amit-jain — APPROVED** [review]
lgtm!

**Comment by rishabhdaim** [comment]
@amit-jain @anchela @mreutegg could you please review the PR, after the new code changes?

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Constants.java [hunk 1]
```diff
@@ -107,11 +107,23 @@ public final class S3Constants {
      */
     public static final String S3_ENCRYPTION_SSE_KMS = "SSE_KMS";
 
+    /**
+     *  Constant to set SSE_KMS encryption.
```
Comment (mreutegg) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Constants.java]: I think this comment is wrong.
```suggestion
     *  Constant to set SSE_C encryption.
```

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Constants.java [hunk 2]
```diff
@@ -107,11 +107,23 @@ public final class S3Constants {
...
+     */
+    public static final String S3_ENCRYPTION_SSE_C = "SSE_C";
+
     /**
      *  Constant to set keyID for SSE_KMS encryption.
      */
     public static final String S3_SSE_KMS_KEYID = "kmsKeyId";
 
+    /**
+     *  Constant to set base64 encoded keyID for SSE_C encryption.
```
Comment (mreutegg) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Constants.java]: I think keyID is misleading. This is the actual customer key.

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Constants.java [hunk 3]
```diff
@@ -107,11 +107,23 @@ public final class S3Constants {
...
+     */
+    // please use  'openssl rand -base64 -out ssec.key 32' command to
+    // generate base64 encoded 32 bytes string customer key for SSE_C
+    public static final String S3_SSE_C_KEYID = "sseCustomerKeyId";
```
Comment (mreutegg) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Constants.java]: I would rename this.
```suggestion
    public static final String S3_SSE_C_KEY = "sseCustomerKey";
```

**Review by mreutegg — CHANGES_REQUESTED** [review]

**Comment by mreutegg** [comment]
See also my [comment](https://issues.apache.org/jira/browse/OAK-10093?focusedCommentId=17732971&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17732971) in JIRA about use of customer provided encryption key and direct binary access feature in Oak.

**Review by mreutegg — APPROVED** [review]
Looks good to me.

**Pull request merged**